docs/api-reference/cli.md

```diff
@@ -106,6 +106,22 @@ PORT=4000 npx next start
 
 > Note: `PORT` can not be set in `.env` as booting up the HTTP server happens before any other code is initialized.
 
+### Working with Load-balancer Timeouts
+
+When deploying Next.js behind an upstream proxy (e.g. a load-balancer like AWS ELB/ALB) it's important to configure Next's underlying http-server with [keep-alive timeouts](https://nodejs.org/api/http.html#http_server_keepalivetimeout) that are _smaller_ than the upstream proxies' timeouts. Otherwise, once a keep-alive timeout is reached for a given TCP connection, Node.js will immediately terminate that connection without notifying the upstream proxies. This results in the upstream proxies throwing intermittent 5xx errors whenever they attempt to reuse a connection that Node.js has already terminated.
```

Comment:
I think it's actually the opposite: Next.js should be configured with a timeout that is _larger_ than the LB's timeout. See here: https://adamcrowder.net/posts/node-express-api-and-aws-alb-502/ We had 502s and set our timeout slightly higher than the LBs and that resolved them for us.

Reply:
See also here: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#connection-idle-timeout

> We also recommend that you configure the idle timeout of your application to be larger than the idle timeout configured for the load balancer. Otherwise, if the application closes the TCP connection to the load balancer ungracefully, the load balancer might send a request to the application before it receives the packet indicating that the connection is closed. If this is the case, the server refuses the request from the load balancer and then the load balancer sends an HTTP 502 Bad Gateway error to the application.

Reply:
yea, you're right lol that's actually what i meant